### PR TITLE
[Filebeat] Improve ECS categorization field mappings for azure module

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?
Improve ECS categorization field mappings for azure module.
Specifically:

- activitylogs
  + add azure.activitylogs.result_type
  + set default_field: false
  + populate event.outcome with allowed values
  + set event.action
  + populate event.category with allowed values
  + set event.kind
  + set event.type
  + add support tickets example
  + add geoip for source.ip
  + add AS info for source.ip
  + add user.name
  + add user.full_name
  + add user.domain
- auditlogs
  + set default_field: false
  + add azure.auditlogs.category
  + populate event.outcome with allowed values
  + set event.action
  + set event.kind
- signinlogs
  + set default_field: false
  + set event.action
  + populate event.category with allowed values
  + set event.type
  + populate event.outcome with allowed values
  + add azure.signinlogs.category
  + add azure.signinlogs.result_type
  + set user.name
  + set user.domain
  + set user.full_name
  + set user.id
  + add geoip for source.ip
  + add AS info for source.ip

## Why is it important?
ECS categorization fields allow cross correlation between filesets.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
```
TESTING_FILEBEAT_MODULES=azure mage -v pythonIntegTest
```

## Related issues
- Closes 16155

### CODE REVIEW

**Comment by andrewkroh** [comment]
run tests

File: x-pack/filebeat/module/azure/signinlogs/test/signinlogs.log-expected.json [hunk 1]
```diff
@@ -34,13 +35,21 @@
...
+        "azure.signinlogs.result_type": "50140",
         "azure.tenant_id": "8a4de8b5-095c-47d0-a96f-a75130c61d53",
         "cloud.provider": "azure",
-        "event.category": "SignInLogs",
+        "event.action": "Sign-in activity",
+        "event.category": [
+            "authentication"
+        ],
         "event.dataset": "azure.signinlogs",
-        "event.duration": 0.0,
```
Comment (andrewkroh) [inline x-pack/filebeat/module/azure/signinlogs/test/signinlogs.log-expected.json:41]: That type checking PR you opened is looking like a very good idea 😄 .

**Comment by leehinman** [comment]
@narph tested on azure and with changes to the dashboards everything looks ok.  anything else you can think of to check?

**Comment by narph** [comment]
> @narph tested on azure and with changes to the dashboards everything looks ok. anything else you can think of to check?

LGTM, thanks

**Comment by kevinserafin** [comment]
Pardon the question but I was attempting to integrate some of these pipeline changes in now and I am wondering if I am not understanding how the error_code set processors are supposed to function.

- set:
    field: event.outcome
    value: success
    if: "ctx?.azure?.signinlogs?.properties?.status?.error_code == null || ctx.azure.signinlogs.properties.status.error_code == 0"
- set:
    field: event.outcome
    value: failure
    if: "ctx?.azure?.signinlogs?.properties?.status?.error_code != null || ctx.azure.signinlogs.properties.status.error_code > 0"

So if I pass a zero for azure.signinlogs.properties.status.error_code, is it not evaluated as true for both the success and failure processor? All of my testing kept events showing up as failures so I started poking around, changing the failure processor to Boolean And seemed to fix it but I wanted to see if I was missing something first.

**Comment by leehinman** [comment]
@kevinserafin you are correct that was a bug.  PR https://github.com/elastic/beats/pull/20254 should fix

**Pull request merged**